File: HackerRank/plusMinus.py
# https://www.hackerrank.com/challenges/plus-minus/problem

def plusMinus(arr):
    pos = len(list(filter(lambda x: (x > 0), arr)))
    neg = len(list(filter(lambda x: (x < 0), arr)))
    zer = len(list(filter(lambda x: (x == 0), arr)))
    
    print("{0:.6f}".format(pos/len(arr)))
    print("{0:.6f}".format(neg/len(arr)))
    print("{0:.6f}".format(zer/len(arr))) 
    
plusMinus([-4, 3, -9, 0, 4, 1])

File: Solutions/HackerRank/plusMinus.c
/* https://www.hackerrank.com/challenges/plus-minus/problem */

#include<stdio.h>
#include<stdlib.h>

void plusMinus(int* array, int length);

int main(){
	
	int array[] = {-4, 3, -9, 0, 4, 1};
	int length = sizeof(array)/sizeof(int);
	
	plusMinus(array, length);
	
	return 0;
}

void plusMinus(int* arr, int arr_count){
	
	int pos=0, neg=0, zer=0, i;
	
	for(i=0; i<arr_count; i++){
		
		if(arr[i]>0) ++pos;
		if(arr[i]==0) ++zer;
		if(arr[i]<0) ++neg;
	}
	
	printf("%.6f\n", (float)pos/arr_count);
	printf("%.6f\n", (float)neg/arr_count);
	printf("%.6f\n", (float)zer/arr_count);
}
